docs/concepts/what-is-a-workflow-execution.md

```diff
@@ -2,57 +2,123 @@
+The Temporal Platform ensures the state of the Workflow Execution persists in the face of failures and outages and resumes execution from the latest state.
+
+**Scalability**
+
+Scalability is responsiveness in the presence of load.
+A single Workflow Execution is limited in size and throughput but is scalable because it can [Continue-As-New](/docs/concepts/what-is-continue-as-new) in response to load.
+A Temporal Application is scalable because the Temporal Platform is capable of supporting millions to billions of Workflow Executions executing concurrently, which is realized by the design and nature of the [Temporal Cluster](/docs/concepts/what-is-a-temporal-cluster) and [Worker Processes](/docs/concepts/what-is-a-worker-process).
+
+### Commands & awaitables
+
+A Workflow Execution does two things:
+
+1. Issue [Commands](/docs/concepts/what-is-a-command)
+2. Wait on an Awaitables (often called Futures)
+
+![Command generation and waiting](/diagrams/workflow-execution-progession-simple.svg)
+
+Commands are issued and Awaitables are provided by the use of Workflow APIs in the [Workflow Definition](/docs/concepts/what-is-a-workflow-definition).
 
-A Workflow Execution has exclusive access to its local state, executes concurrently to all other Workflow Executions, and can communicate with other Workflow Executions using Signals.
+Commands are generated whenever the Workflow Function is executed.
+The Worker Process supervises the Command generation and makes sure that it maps to the current Event History.
+(For more information, see [Deterministic constraints](/docs/concepts/what-is-a-workflow-definition/#deterministic-constraints).)
+The Worker Process batches the Commands and then suspends progress to send the Commands to the Cluster whenever the Workflow Function reaches a place where it can no longer progress without a result from a Awaitable.
 
-A Workflow Execution is either Running or Closed.
-When a Workflow Execution is Running, it is either actively progressing or suspended, awaiting on something.
+A Workflow Execution may only ever block progress on an Awaitable that is provided through a Temporal SDK API.
+Awaitables are provided when using APIs for the following:
 
-![Workflow Execution Running status](/diagrams/workflow-execution-running-status.svg)
+- Spawning a [Child Workflow Execution](/docs/concepts/what-is-a-child-workflow-execution): Progress can block on confirmation that the Child Workflow Execution started, and on the result of the Child Workflow Execution.
+- Sending a [Signal](/docs/concepts/what-is-a-signal): Progress can block on confirmation that the Signal sent.
+- Spawning an [Activity Execution](/docs/concepts/what-is-an-activity-execution): Progress can block on the result of the Activity Execution.
+- Requesting cancellation of another Workflow Execution: Progress can block on confirmation that the other Workflow Execution is cancelled.
```

Comment:
@tsurdilo I made a guess on this one. Can you confirm?
Or can it block on confirmation that the request to cancel is sent - or both?